### PR TITLE
LastModifiedByAwareDAO

### PR DESCRIPTION
**Pull request by kirkeaton**
- Made some concurrency changes to JDAO and FileJournal
 - Wrote LastModifiedByAwareDAO

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/auth/LastModifiedByAwareDAO.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  name: 'LastModifiedByAwareDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        if ( obj instanceof LastModifiedByAware ) {
+          User user = (User) x.get("user");
+          ((LastModifiedByAware) obj).setLastModifiedBy(user.getId());
```
Comment (jlhughes) [inline src/foam/nanos/auth/LastModifiedByAwareDAO.js:18]: Should we be cloning here? (fclone)

**Comment by eatonk** [comment]
Why do we need to clone here?

File: src/foam/nanos/auth/LastModifiedByAwareDAO.js (hunk 1 above)
Comment (kirkeaton) [reply src/foam/nanos/auth/LastModifiedByAwareDAO.js:18]: Why do we need to clone here?

Comment (jlhughes) [reply src/foam/nanos/auth/LastModifiedByAwareDAO.js:18]: ah yes, it's already been cloned.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**